controllers/feeds/getFeedsV2.js

```diff
@@ -254,45 +295,27 @@ module.exports = async (req, res) => {
       }
     }
     // get karma score for each post actor
-    for (let i = 0; i < data.length; i++) {
-      if (data[i].latest_reactions?.downvotes) {
-        for (let j = 0; j < data[i].latest_reactions?.downvotes.length; j++) {
-          if (!postActors.includes(data[i].latest_reactions.downvotes[j].user_id)) {
-            postActors.push(data[i].latest_reactions.downvotes[j].user_id);
-          }
-        }
-      }
-      if (data[i].own_reactions?.downvotes) {
-        for (let j = 0; j < data[i].own_reactions?.downvotes.length; j++) {
-          if (!postActors.includes(data[i].own_reactions.downvotes[j].user_id)) {
-            postActors.push(data[i].own_reactions.downvotes[j].user_id);
-          }
-        }
-      }
+    for (const d of data) {
+      const actor_from_reaction = getActorFromLatestReaction(d.latest_reactions);
+      postActors = [...postActors, ...actor_from_reaction];
     }
 
     const karmaScores = await UsersFunction.getUsersKarmaScore(User, postActors);
     for (let i = 0; i < data.length; i++) {
       const user = karmaScores.find((user) => user.user_id === data[i].actor.id);
       data[i].karma_score = roundingKarmaScore(user?.karma_score || 0);
-      if (data[i].latest_reactions?.downvotes) {
-        data[i].latest_reactions.downvotes = data[i].latest_reactions?.downvotes.map((downvote) => {
-          const user = karmaScores.find((user) => user.user_id === downvote.user_id);
-          return {
-            ...downvote,
-            karma_score: roundingKarmaScore(user?.karma_score || 0)
-          };
-        });
+
+      if (data[i].latest_reactions) {
+        data[i].latest_reactions = addKarmaScoreToLatestReaction(
+          data[i].latest_reactions,
+          karmaScores
+        );
       }
-      if (data[i].own_reactions?.downvotes) {
-        data[i].own_reactions.downvotes = data[i].own_reactions?.downvotes.map((downvote) => {
-          const user = karmaScores.find((user) => user.user_id === downvote.user_id);
-          return {
-            ...downvote,
-            karma_score: roundingKarmaScore(user?.karma_score || 0)
-          };
-        });
+
+      if (data[i].own_reactions) {
+        data[i].own_reactions = addKarmaScoreToLatestReaction(data[i].own_reactions, karmaScores);
       }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This section of code is retrieving karma scores for each post actor and adding them to the data. The logic seems fine, but there's a potential performance issue here. You're calling `getUsersKarmaScore` once for all actors, which could be a large number of database queries if there are many unique actors. Consider optimizing this by batching the queries or using a more efficient data retrieval method if possible.

Also, similar to the previous comments, you're directly modifying the `data[i]` objects inside the loop. Consider creating new objects instead of mutating the existing ones to avoid potential side effects.


<!-- This is an auto-generated comment by OSS CodeRabbit -->